### ISSUE TITLE
Missing/Mismatched Emojis

### ISSUE THREAD

**Issue by jake770321** [getsolus/packages#4181]
### Please confirm there isn't an existing open bug report

- [X] I have searched open bugs for this issue

### Summary

Some emojis are missing in certain applications. It seems that the "NotoColorEmoji.ttf" included with solus doesn't show all emojis properly. The filesize for the one included with solus is 10.6mb whereas the one downloaded from [https://fonts.google.com/noto/specimen/Noto+Color+Emoji](url) is 24.3mb. Downloading and replacing the included NotoColorEmoji.ttf file with the one downloaded from the link above seems to fix the issue.

### Steps to reproduce

1. Using the steam client as an example: Try and post the following emoji ⚔
It only shows a very simplified version.

2. Now replace the default "NotoColorEmoji.ttf" at /usr/share/fonts/truetype/noto-sans-ttf with the one downloaded from [https://fonts.google.com/noto/specimen/Noto+Color+Emoji](url) and reboot.

3. The emoji is now shown properly in full colour. 

### Expected result

The emoji being shown properly.

### Actual result

Seeing a very simplified version of only certain emojis leading to a mismatch. 

### Environment

- [X] Is system up to date?

### Repo

Shannon (stable)

### Desktop Environment

Budgie

### System details

Budgie Version: 10.9.2
OS Name: Solus 4.6 Convergence
OS Type: 64-bit
Windowing System: X11
Kernel Version: Linux 6.11.5-307.current

**Comment by malfisya** [comment]
@Staudey Any idea?

**Comment by Staudey** [comment]
I've been experimenting with merging the different font files contained in noto-emoji, but don't have a proper result yet. Will look into it again.

**Comment by Staudey** [comment]
Hmm, @jake770321, can you clarify how you tested this a bit more? When I replace our `NotoColorEmoji.ttf` with the font file downloaded from the page you linked, I get the exact same result for the ⚔ emoji in chat windows / reviews / discussions as before (that is, the simplified version). Did you do anything else?
I'll have to redo some of my packaging changes after confirming a proper way to test this, because so far I've only tried the Steam chat window but until now didn't realise that it didn't use a different emoji type even with the downloaded font file.

**Comment by jake770321** [comment]
> Hmm, @jake770321, can you clarify how you tested this a bit more? When I replace our `NotoColorEmoji.ttf` with the font file downloaded from the page you linked, I get the exact same result for the ⚔ emoji in chat windows / reviews / discussions as before (that is, the simplified version). Did you do anything else? I'll have to redo some of my packaging changes after confirming a proper way to test this, because so far I've only tried the Steam chat window but until now didn't realise that it didn't use a different emoji type even with the downloaded font file.

Very weird. I'm having the same issue now with that emoji regardless of whether I replace the file but not with some other emojis. 
Try the steps above but with the following emoji: 🗣
Reboot after replacing the file and try posting it in a chat window

**Comment by Staudey** [comment]
Thanks! I now have something that's at least a partial solution to the problem.

(Weirdly though it didn't work for `firefox`, where it showed the coloured emoji for a second after launching and then reverted to the monocolour one)

**Comment by ReillyBrogan** [comment]
I've reverted the commit for this since it completely broke emoji rendering in Qt apps, so re-opening.

**Comment by ReillyBrogan** [comment]
I would note that I took a look at the Fedora and Arch packages and neither of them ship this file, is the issue reproducible on those distributions as well? If not then it might be a configuration issue elsewhere.

**Comment by ReillyBrogan** [comment]
I tried to reproduce this issue and I can't. The "speaking head" emoji works fine in both Firefox and Steam chat, both the uncolored (🗣) and the colored (🗣️) variants.

**Comment by Staudey** [comment]
Yeah, this is about showing the standard emojis in colour (without variation selector), as it does on e.g. Android, and I assume Windows and other systems.

**Comment by Staudey** [comment]
Output on Solus:

![](https://i.imgur.com/HT1QLiR.png)

Output in Android app:

![](https://i.imgur.com/WkTYpI1.png)

**Comment by Staudey** [comment]
It does however NOT print the ⚔ in colour without explicitly setting the variation, so I dunno what the actual logic behind it is

**Comment by Staudey** [comment]
Finding some posts about how there is no rhyme or reason behind it :man_shrugging: 
https://www.codejam.info/2021/11/emoji-variation-selector.html

**Comment by ReillyBrogan** [comment]
So this is actually a problem space I'm (unfortunately) quite familiar with. Basically, the logic for when to "upgrade" monochrome emoji to colored ones is actually quite complex and is something that the application itself needs to implement. You can't hack this unfortunately, any possible workaround you can come up with will unfortunately break something else (see our former fontconfig hack that broke certain websites). Qt recently had to import code from chromium in order to get this working for example (which I have backported to our Qt 6.8 packages): https://bugreports.qt.io/browse/QTBUG-85744. I know that Steam uses CEF (Chromium Embedded Framework) which possibly should support this, but perhaps it's not wired up correctly or otherwise not used for the chat windows. Most likely this is an upstream issue and the Steam client will need to be updated by Steam devs.

I'm going to mark this one as an upstream issue, but please test if the issue appears on other distributions as well. If it doesn't then we can look into it further. Otherwise we can close this as there's nothing we can really do here.

**Comment by TraceyC77** [comment]
Closing since the requested information wasn't provided
